File: codeeval/js/email_validation.js
/*
 *
 * Email Validation - Code Eval Challenge
 * Tyler Childs - 12/12/2013
 *
 * */

var fs  = require("fs");

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
	if (line !== "") {
		if(/^[a-zA-Z0-9.+-_]*@[a-zA-Z0-9.-]*\.[a-zA-Z0-9]{2,4}$/.test(line))
			console.log("true");
		else
			console.log("false");
	}
});

File: codeeval/ruby/email_validation.rb
#
#
# Email Validation - Code Eval Challenge
# Tyler Childs - 10/12/2014
#

# open file from command line
if ARGV[0].nil?
	puts "\nError: No file specified\n\n"
	exit 1
else
	file = File.read(ARGV[0])
end

def validate_email(string)
	return false if /^[a-zA-Z0-9.+\-_]*@[a-zA-Z0-9.\-]*\.[a-zA-Z0-9]{2,4}$/.match(string).nil?

	true
end

# loop through file line by line
file.each_line do |line|

	puts validate_email(line)

end

exit 0